src/ParseServer.js

```diff
@@ -171,7 +177,11 @@ class ParseServer {
     const userController = new UserController(emailControllerAdapter, appId, { verifyUserEmails });
     const liveQueryController = new LiveQueryController(liveQuery);
 
+    if (settingsCacheOptions) {
+      cache.apps = PersistentSettingsStore(settingsCacheOptions, definedOptions);
```

Comment:
the other problem is the risk of error, rewriting from the passed options, and losing the changes (like masterKey) that has been changed from the dashboard.